### DISCUSSION TITLE
Is it possible to add an image as background instead of black screen?

### DISCUSSION THREAD

**Discussion by rezpower** [tomasklaen/uosc#589] [Q&A] [answered]
I was wondering if it's possible to add a wallpaper to load instead of the black screen when you open MPV? This is useful on startup, and also when playing music with no image. 
If already possible please tell me how to achieve this :)

**Comment by christoph-heinrich** [comment]
There is [`--background=`](https://mpv.io/manual/master/#options-background) for specifying the background color, but setting an actual image as background is not possible afaik.

**Reply by rezpower** [reply]
Thanks a lot for your reply. I now got the answer to my question, BUT it would really be great to add this option if implementation is not too complicated.

**Reply by christoph-heinrich** [reply, marked as answer]
While it would be technically possible for us to draw some image by converting it to raw pixels and using [overlay-add](https://mpv.io/manual/master/#command-interface-overlay-add), that's a terrible way of going about that.
Such a feature should really be implemented in mpv itself, so please make a feature request [over there](https://github.com/mpv-player/mpv/issues) instead.

**Comment by rezpower** [comment]
Thank you very much. I will do that :)

**Comment by AndyM48** [comment]
If you have `keep-open=yes` and a valid geometry such as `geometry=50%x50%+50%+50%` in your mpv.conf file then you can open mpv with an image instead of a blank screen. It helps to have the image file in your favourite directory such as ~/Videos, then the uosc will open files from that directory first.

mpv-coverart from github can use a default image if no cover art exists. (requires `force-window=yes`)

**Comment by horous** [comment]
Hi body, could share your solution here? I am also confused by this problem for a long time

**Comment by AndyM48** [comment]
1) Start mpv with an image instead of a blank screen: Edit mpv.conf, add `keep-open=yes` and a valid geometry such as `geometry=50%x50%+50%+50%` now open mvp with `mpv /path/to/folder/image_filename` for example 
`$ mpv /home/user/Videos/mpv_background.jpg`

2) Show image when when playing music with no image: Copy coverart.lua to your scripts directory, copy coverart.conf to your script.opts directory, add `force-window=yes' to mpv.conf, edit coverart.conf  and add your default cover art image to `placeholder=`
ref: https://github.com/CogentRedTester/mpv-coverart

**Reply by horous** [reply]
aha, thank you very much. But i am not a programmer, it is difficult for me to do this. Thank you anyway.
